src/scope/future/README.md

```diff
@@ -0,0 +1,267 @@
+found in `F` anymore.
+
+This is good, because there *are* still active refs to the
+`ScopeFuture` after we enter the *COMPLETE* state. There are two
+sources of these: unpark values and the future result.
+
+**Unpark values.** We may have given away `Arc<Unpark>` values --
+these are trait objects, but they are actually refs to our
+`ScopeFuture`. Note that `Arc<Unpark>: 'static`, so these could be
+floating about for any length of time (we had to transmute away the
+lifetimes to give them out). This is ok because (a) the `Arc` keeps
+the `ScopeFuture` alive and (b) the only thing you can do is to call
+`unpark()`, which will promptly return since the state is *COMPLETE*
+(and, anyhow, as we saw above, it doesn't have access to any
+references anyhow).
+
+**Future result.** The other, more interesting reference to the
+`ScopeFuture` is the value that we gave back to the user when we
+spawned the future in the first place. This value is more interesting
+because it can be used to do non-trivial things, unlike the
+`Arc<Unpark>`. If you look carefully at this handle, you will see that
+its type has been designed to hide the type `F`. In fact, it only
+reveals the types `T` and `E` which are the ok/err result types of the
+future `F`.  This is intentonal: suppose that the type `F` includes
+some references, but those references don't appear in the result. We
+want the "result" future to be able to escape the scope, then, to any
+place where the types `T` and `E` are still in scope. If we exposed
+`F` here that would not be possible. (Hiding `F` also requires a
+transmute to an object type, in this case an internal trait called
+`ScopeFutureTrait`.) Note though that it is possible for `T` and `E`
+to have references in them. They could even be references tied to the
+scope.
+
+So what can a user do with this result future? They have two
+operations available: poll and cancel. Let's look at cancel first,
+since it's simpler. If the state is *COMPLETE*, then `cancel()` is an
+immediate no-op, so we know that it can't be used to access any
+references that may be invalid. In any case, the only thing it does is
+to set a field to true and invoke `unpark()`, and we already examined
+the possible effects of `unpark()` in the previous section.q
+
+So what about `poll()`? This is how the user gets the final result out
+of the future. The important thing that it does is to access (and
+effectively nullify) the field `result`, which stores the result of
+the future and hence may have access to `T` and `E` values. These
+values may contain references...so how we know that they are still in
+scope?  The answer is that those types are exposed in the user's type
+of the future, and hence the basic Rust type system should guarantee
+that any references are still valid, or else the user shouldn't be
+able to call `poll()`. (The same is true at the time of cancellation,
+but that's not important, since `cancel()` doesn't do anything of
+interest.)
```

Comment:
This makes me nervous, as it probably should since you bothered to document it so carefully.  But I don't have any concrete objection, and I trust your intuition on this more than my own, so... :man_shrugging: